tensorflow_model_optimization/python/core/sparsity_tf2/riglpruner_test.py

```diff
@@ -0,0 +1,577 @@
+  grad = tf.random.shuffle(np.linspace(1., 100., n))
+  return tf.cast(grad, base_type)
+
+class RiglPruningTest(test.TestCase, parameterized.TestCase):
+
+  def setUp(self):
+    super(RiglPruningTest, self).setUp()
+    self.block_size = (1, 1)
+    self.block_pooling_type = "AVG"
+    self.target_sparsity = 0.5
+    self.initial_drop_fraction = 0.3 # i.e. 0.3 * target sparsity per layer, annealed by schedule below
+    self.constant_updater = schedule.ConstantSchedule(self.initial_drop_fraction, 0, 100, 1)
+    self.skip_updater = schedule.ConstantSchedule(self.initial_drop_fraction, 0, 100, 2)
+    self.updater = schedule.ConstantSchedule
+    self.grad = _dummy_gradient
+    self.seed = 0
+    self.noise_std = 1
+    self.reinit = False
+    self.grow_init_zeros = 'zeros'
+    self.grow_init_randn = 'random normal'
+    self.grow_init_randunif = 'random uniform'
+
+  def testGetGrowGrads(self):
+    mask = tf.ones((100))
+    grads = -1 * tf.linspace(1., 6., 6)
+    expected_grads = tf.linspace(1., 6., 6)
+
+    p = pruner.RiGLPruner(
+      update_schedule=self.skip_updater,
+      sparsity=self.target_sparsity,
+      block_size=self.block_size,
+      block_pooling_type=self.block_pooling_type,
+      seed=self.seed,
+      noise_std=self.noise_std,
+      reinit=self.reinit,
+      grow_init=self.grow_init_zeros
+    )
+
+    grow_grads = p._get_grow_grads(mask, grads)
+    self.assertAllEqual(grow_grads, expected_grads)
+
+
+  def testDropandGrowMasksDisjoint(self):
+    weight = tf.Variable(np.linspace(1.0, 100.0, 100))
+    weight_dtype = weight.dtype.base_dtype
+    mask = tf.Variable(
+        tf.ones(weight.get_shape(), dtype=weight_dtype),
+        dtype=weight_dtype)
+    grad = self.grad(weight)
+
+    p = pruner.RiGLPruner(
+      update_schedule=self.skip_updater,
+      sparsity=self.target_sparsity,
+      block_size=self.block_size,
+      block_pooling_type=self.block_pooling_type,
+      seed=self.seed,
+      noise_std=self.noise_std,
+      reinit=self.reinit
+    )
+
+    optimizer = tf.keras.optimizers.SGD(learning_rate=0.01)
+    optimizer.iterations.assign(0)
+    p.create_slots(optimizer, weight)
+
+    n_prune = 25 # drop ratio 0.5 of 50% sparse
+    n_total = 100
+    n_keep = n_total - n_prune
+    current_mask = optimizer.get_slot(weight, 'mask').read_value()
+    dropped_mask = p._generic_top_k(weight * current_mask, current_mask, n_keep,  n_total)
+    grow_scores = tf.math.abs(grad)
+    grow_scores = tf.reshape(grow_scores, (-1,))
+    # set enabled connections (ones) to min(scores) - 1, i.e. they have the lowest scores
+    grow_scores_lifted = tf.where(
+      tf.math.equal(dropped_mask, 1),
+      tf.ones_like(dropped_mask) * (tf.reduce_min(grow_scores) - 1), grow_scores
+    )
+    grown_mask = p._generic_top_k(grow_scores_lifted, current_mask, n_prune,  n_total)
+    self.assertAllEqual(tf.reduce_sum(dropped_mask * grown_mask), 0.)
+
+
+  def testMaskNoChangeBeforeandAfter(self):
```

Comment:
Is this a test that's supposed to check that there was no update on a step that has updating disabled? If so can the test name + a comment clarify this?

If not, what about this test makes it so that we expect no updates? It needs to be clear